languageserver/integration/codelenses.go

```diff
@@ -83,23 +85,45 @@ func (i *FlowIntegration) showDeployContractAction(
 	}
 
 	name := contract.Identifier.Identifier
-
 	position := contract.StartPosition()
+	codelensRange := conversion.ASTToProtocolRange(position, position)
+	var codeLenses []*protocol.CodeLens
+
+	// If emulator is not up, we need to provide actionless codelens
+	if i.emulatorState == EmulatorStarting || i.emulatorState == EmulatorOffline {
+		return codeLenses
+	}
+
+	signersList := entryPointInfo.pragmaSignersStrings[:]
+	// TODO: resolve check for amount of signers equals to provided by pragma
+	if len(signersList) == 0 {
+		activeAccount := [][]string{[]string{"active account"}}
+		signersList = append(signersList, activeAccount...)
+	}
+
+	for _, signers := range signersList {
+		title := fmt.Sprintf(
+			"%s %s %s %s %s",
+			prefixOK,
+			"Deploy contract",
+			name,
+			"to",
+			signers[0],
```

Comment:
You can embed the constant parts directly in the format string:

```suggestion
			"%s Deploy contract %s to %s",
			prefixOK,
			name,
			signers[0],
```